packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -183,6 +183,12 @@ class FlutterCommandRunner extends CommandRunner<void> {
     try {
       if (platform.script.scheme == 'data')
         return '../..'; // we're running as a test
+
+      if (platform.script.scheme == 'package') {
+        String packageConfigPath = Uri.parse(platform.packageConfig).toFilePath();
```

Comment:
Perhaps give an error if packageConfig is null (when no `--packages` flag is specified)?

Reply:
If there is no `packageConfig`, then Dart would not have been able to launch `flutter_tools` if a `package:` URI was specified on the Dart command line